From 2f88b67f7969655e222d2c26297b02add47c1005 Mon Sep 17 00:00:00 2001
From: Manjunath <manjunathx.b.s@intel.com>
Date: Mon, 3 Dec 2012 15:58:44 +0530
Subject: Can't play AAC file with ID3 tag.

BZ: 72082

When the file is scanned, the ID3 tag is found to have a
footer at the end of audio data. And getAdtsFrameLength
return zero after reading the footer. As a result, the
duration and mInitCheck are not set. The current change
considers the data till footer and sets the duration based
on no. of frames read till the footer/corrupt frame.

Category: aosp improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: Yes

Orig-Change-Id: Ib362f884cd72a8624de26823956b312c886f3030

Change-Id: I0eae71c0c2906a9d918b5e6edaf5a96f7a8029c7
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>
Reviewed-on: http://android.intel.com:8080/82061
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libstagefright/AACExtractor.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/AACExtractor.cpp b/media/libstagefright/AACExtractor.cpp
index 0fb7bbf..041cd58 100644
--- a/media/libstagefright/AACExtractor.cpp
+++ b/media/libstagefright/AACExtractor.cpp
@@ -174,7 +174,7 @@ AACExtractor::AACExtractor(
     if (mDataSource->getSize(&streamSize) == OK) {
          while (offset < streamSize) {
             if ((frameSize = getAdtsFrameLength(source, offset, NULL)) == 0) {
-                return;
+                break;
             }
 
             mOffsetVector.push(offset);
-- 
1.7.4.1

